src/streamlink_cli/output.py

```diff
@@ -140,8 +140,47 @@ def _create_arguments(self):
 
             # mpv
             if self.player == "mpv":
-                # see https://mpv.io/manual/stable/#property-expansion, allow escaping with \$
-                self.title = self.title.replace("$", "$$").replace("\$$", "$")
+                # see https://mpv.io/manual/stable/#property-expansion, allow escaping with \$, respect mpv's $>
+                ##mpv has a "disable property-expansion" token which must be handled in order to accurately represent $$ in title
+                if '\$>' in self.title:
```

Comment:
Need some tests for this :)

Reply:
there are some! if you want to write more, that'd be cool. :) not sure what more I would add.

Reply:
True, I didn't see them hiding there. I would probably separate this in to a separate method as it is quite complex, would make it easier to test standalone and possibly refactor.

Reply:
It could be put into a separate function, yeah, as there are a lot of branches and complexity to disabling $ doubling partway through a string depending on how many leading $ are in front of a > & still replacing \$. I tried a different implementation before parsing the string one character at a time, but it failed on some tests and actually wasn't much "better to read"/"more maintainable" than this version... :)

What file should a refactored "mpv format title string" method go into?

Reply:
I would keep it as a method of `Player` for now, `_mpv_title_escape` or something. Later I was thinking we could refactor `Player` so that subclasses are possible, like `PlayerMPV` and `PlayerVLC` to handle the specifics of those players. But that would like require a factory method for `Player` or some `__new__` shenanigans.

Reply:
Are the idea of these transformations to stop Twitch channels from putting formatting commands in the titles on their streams? Like if I set the title of my stream to 'You're using MPV v${mpv-version}', it wouldn't render the MPV version, but the literal string?

Reply:
There's really no way to stop channels from putting format commands in the titles of their streams, unless you want to look for them and strip them before LazyFormatter puts them into PlayerOutput's self.title. The format codes are harmless and can't execute naughty code, it could just cause a desync of what is displayed in webbrowser at twitch.tv and in streamlink's window title. Because there's no motivation for evil hackers to insert format codes, and because streamers must put the substring "\$" in their title to activate them, it's more likely a "feature" than a "bug" that streamers could put player specific format codes in their stream title.

The idea of `_mpv_title_escape' is to render multiple `$` in a row correctly, to reduce accidentally triggering format codes, and to enable triggering format codes when desired. VLC can do it in a one-liner, but mpv has that annoying "disable format codes from this point on" token.

Reply:
The only reason I asked it because I couldn't get it to work ... but now it does ;)

Reply:
`--title '{title} (\${{mpv-version}})'` works as expected with mpv :)